src/oauth.rs

```diff
@@ -0,0 +1,88 @@
+use crate::settings;
+use crate::token::{verify_jwt_token_from_rsa, Claims};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug)]
+pub struct Response {
+    pub email: String,
+    pub claims: Claims,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Key {
+    // kty: String,
+    pub n: String, // We only need n and e
+    pub e: String,
+    // qi: String,
+    // kid: String,
+    // alg: String,
+    // r#use: String,
```

Comment:
Which of these fields we have to keep with ourselves?

Reply:
As you note, we only need `n` and `e` here. The other fields can be safely skipped for now. Since `serde_json` is part of `serde`, you can use some of the [field attributes](https://serde.rs/field-attrs.html) described here. They will let you specify fields you can `skip`, or `rename`.